internal/service/order/service.go

```diff
@@ -0,0 +1,150 @@
+package order
+
+import (
+	"context"
+	"errors"
+	"log/slog"
+
+	"github.com/jackc/pgx/v5"
+
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/ktigay/loyalty/internal/entity"
+	repo "github.com/ktigay/loyalty/internal/repository/order"
+)
+
+var (
+	ErrWrongOrderNumber   = errors.New("wrong order number")
+	ErrWrongUser          = errors.New("wrong user order")
+	ErrOrderAlreadyExists = errors.New("order already exists")
+)
+
+// AccrualClientInterface Интерфейс сервиса Accrual.
+//
+//go:generate mockgen -destination=./mocks/mock_accrual.go -package=mocks github.com/ktigay/loyalty/internal/service/order AccrualClientInterface
+type AccrualClientInterface interface {
+	OrdersStatus(ctx context.Context, orderIDs []string) (*[]entity.Accrual, error)
+}
```

Comment:
Сразу несколько рекомендаций здесь:
- Не стоит а интерфейсах использовать слово Interface. Это не добавляет дополнительного контекста при этом делает наименование длинным. В Go такая нотация не используется.
- Я бы советовал назвать метод так чтобы он давал подсказку о том, что он делает и возвращает - возвращает он заказы, значит `GetOrder(...)`. 
- Интерфейс с одним методом стоит называть по имени этого метода + суффикс er - OrdersGetter, например. https://go.dev/doc/effective_go#interfaces
```suggestion
type OrdersGetter interface {
	GetOrders(ctx context.Context, ids ...string) ([]entity.Accrual, error)
}
```

Reply:
Я несколько раз встречал рекомендации не использовать префикс Get в названии методов. Может быть это касается только каких-то определенных методов? Например там где у структуры геттеры и сеттеры

Reply:
Действительно, для геттеров не рекомендуется использовать префикс `Get` https://go.dev/doc/effective_go#Getters. В данном случае наверное тоже можно без него обойтись, с другой стороны он вроде бы тут и не мешает потому что подчеркивает то, что метод инициирует получение заказа. Может быть можно было бы назвать его `RetrieveOrders` или `RetrieveOrdersWithStatus, если Get ассоциируется с гетером.